### PR TITLE
Retry job on failure to get aws credentials

### PR DESCRIPTION
**Pull request by usamasaqib**
### What does this PR do?
This PR adds a retry rule for when it fails to get aws credentials, such as this error:
https://gitlab.ddbuild.io/DataDog/datadog-agent/-/jobs/520001464

### CODE REVIEW

**Review by paulcacheux — APPROVED** [review]

**Review by gjulianm — APPROVED** [review]

File: tools/ci/aws_ssm_get_wrapper.sh [hunk 1]
```diff
@@ -19,4 +19,4 @@ while [[ $retry_count -lt $max_retries ]]; do
 done
 
 echo "Failed to retrieve parameter after $max_retries retries"
-exit 1
+exit 666
```
Comment (KevinFairise2) [inline tools/ci/aws_ssm_get_wrapper.sh:22]: Did you test that it was working as expected?
If I remember correctly some time ago GitlabCI script was always exiting with 1 even if your script was explicitly exiting with another exit code. That might have been fixed. But I am not 100% sure

Comment (usamasaqib) [reply tools/ci/aws_ssm_get_wrapper.sh:22]: I did not test this. Let me give it a shot

Comment (usamasaqib) [reply tools/ci/aws_ssm_get_wrapper.sh:22]: You're right does not seem to work :(

Comment (brycekahle) [reply tools/ci/aws_ssm_get_wrapper.sh:22]: Turns out this feature was added in 16.11, which we are not running yet.

**Pull request closed without merging**